### PR TITLE
Add support for filtering within arrays

### PR DESCRIPTION
**Pull request by zikaari**

### CODE REVIEW

**Comment by silesky** [comment]
@zikaari I assume you'll add a changeset per our offline discussion 😄

File: packages/browser/package.json [hunk 1]
```diff
@@ -51,7 +51,7 @@
     "@segment/analytics-core": "1.2.5",
     "@segment/analytics.js-video-plugins": "^0.2.1",
     "@segment/facade": "^3.4.9",
-    "@segment/tsub": "1.0.1",
```
Comment (chrisradek) [inline packages/browser/package.json:54]: I'm not sure why we were pegged to a specific version - any reason not to pin to `^2.0.0`?

Comment (zikaari) [reply packages/browser/package.json:54]: I don't know why it was like this, historically. I can make it `^2.0.0` now

**Review by silesky — APPROVED** [review]
🔥

**Pull request merged**